### DISCUSSION TITLE
sqlalchemy issues with poetry ?

### DISCUSSION THREAD

**Discussion by sandys** [sqlalchemy/sqlalchemy#6925] [Usage Questions] [answered]
https://stackoverflow.com/questions/68851505/installing-sqlalchemy-with-poetry-causes-an-attributeerrorr

im seeing the same behavior . Anyone know the cause of this ?

any format like 
SQLAlchemy = "^1.4.22" or SQLAlchemy = ">1.4.22" throws the same error
```

-> % poetry update
Updating dependencies
Resolving dependencies... (1.5s)

Writing lock file

  AttributeError

  'EmptyConstraint' object has no attribute 'allows'

  at ~/Code/miniconda3/lib/python3.9/site-packages/poetry/core/version/markers.py:291 in validate
      287│
      288│         if self._name not in environment:
      289│             return True
      290│
    → 291│         return self._constraint.allows(self._parser(environment[self._name]))
      292│
      293│     def without_extras(self):  # type: () -> MarkerTypes
      294│         return self.exclude("extra")
```

**Comment by CaselIT** [comment, marked as answer]
Hi,

please see #6903 and #6907

tldr: update poetry
